includes/board.h

```diff
@@ -13,18 +13,19 @@ class Board {
     };
 
     struct BoardCell {
-        Block& block;
+        Block* block;
         bool isOccupied;
-        BoardCell(Block& b, bool state = false) : block(b) {
-            this->isOccupied = state;
+        BoardCell() {
+            isOccupied = false;
+            block = new Block();
         }
-        void setOccupied() {
-            this->isOccupied = true;
+        BoardCell(Block& b, bool state = false) : block(&b) {
```

Comment:
I suggest you have the user pass in a pointer rather than a reference but either way it looks like you aren't even using this constructor so might as well remove.